lang/rust/avro/src/schema.rs

```diff
@@ -225,30 +228,31 @@ impl<'a> From<&'a types::Value> for SchemaKind {
 ///
 /// More information about schema names can be found in the
 /// [Avro specification](https://avro.apache.org/docs/current/spec.html#names)
-#[derive(Clone, Debug, Deserialize)]
+#[derive(Clone, Debug, Deserialize, Hash, PartialEq, Eq)]
 pub struct Name {
     pub name: String,
-    pub namespace: Option<String>,
-    pub aliases: Option<Vec<String>>,
+    pub namespace: Namespace,
 }
 
 /// Represents documentation for complex Avro schemas.
 pub type Documentation = Option<String>;
+/// Represents the aliases for Named Schema
+pub type Aliases = Option<Vec<String>>;
+/// Represents Schema lookup within a schema env
+pub type Names = HashMap<Name, Schema>;
```

Comment:
`pub(crate)` ?
Does this need to be visible outside of the crate ?

Reply:
Probably not? I use the same type inside of latest avro derive and would want to use it there, but thats probably the same crate? (I don't know enough about crate packaging to say for certain). I'll make it pub(crate) and change it if needed in avro derive, but then then the change is then obviously associate with the new out of crate use.